src/test/java/seedu/address/security/PasswordManagerTest.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.security;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PasswordManagerTest {
+
+    private static final String TEST_PASSWORD = "TestPassword123!";
+    private static final String PASSWORD_FILE = "password.txt";
+
+    @BeforeEach
+    void setUp() {
+        // Ensure the password file is deleted before each test
+        try {
+            Files.deleteIfExists(Paths.get(PASSWORD_FILE));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @AfterEach
+    void tearDown() {
+        // Clean up the password file after each test
+        try {
+            Files.deleteIfExists(Paths.get(PASSWORD_FILE));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    void testSavePassword() {
+        PasswordManager.savePassword(TEST_PASSWORD);
+
+        // Verify the password is saved correctly
+        String savedHash = PasswordManager.readPassword();
+        assertNotNull(savedHash);
+        assertTrue(savedHash.contains(":")); // Check if hash format is correct
+    }
+
+    @Test
+    void testReadPasswordNoPasswordSet() {
```

Comment:
You could standardise test naming convention to:
`featureUnderTest_testScenario_expectedBehavior()`
From: [https://se-education.org/guides/conventions/java/intermediate.html]((https://se-education.org/guides/conventions/java/intermediate.html))